src/main/java/uk/gov/companieshouse/confirmationstatementapi/eligibility/impl/CompanyTradedStatusValidation.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.companieshouse.confirmationstatementapi.eligibility.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.confirmationstatementapi.eligibility.EligibilityRule;
+import uk.gov.companieshouse.confirmationstatementapi.eligibility.EligibilityStatusCode;
+import uk.gov.companieshouse.confirmationstatementapi.exception.EligibilityException;
+import uk.gov.companieshouse.confirmationstatementapi.model.CompanyTradedStatusType;
+import uk.gov.companieshouse.confirmationstatementapi.service.CorporateBodyService;
+
+public class CompanyTradedStatusValidation implements EligibilityRule<CompanyProfileApi> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CompanyTradedStatusValidation.class);
+
+    private final CorporateBodyService corporateBodyService;
+
+    public CompanyTradedStatusValidation(CorporateBodyService corporateBodyService) {
+        this.corporateBodyService = corporateBodyService;
+    }
+
+    @Override
+    public void validate(CompanyProfileApi profileToValidate) throws EligibilityException {
+        var companyNumber = profileToValidate.getCompanyNumber();
+        LOGGER.info("Validating Company Traded Status for: {}", companyNumber);
+
+        var companyTradedStatus = corporateBodyService.getCompanyTradedStatus(companyNumber);
+
+        if(companyTradedStatus != CompanyTradedStatusType.NOT_ADMITTED_TO_TRADING) {
```

Comment:
Could you reverse these two variables?
`CompanyTradedStatusType.NOT_ADMITTED_TO_TRADING != companyTradedStatus`

Doing it this way round means it is impossible to get a Null pointer error as we can guarantee that NOT_ADMITTED_TO_TRADING exists, but we cannot absolutely guarentee companyTradedStatus isn't null